libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackClientWrapper.cs

```diff
@@ -0,0 +1,554 @@
+        /// Wraps Slack API's GetCountsAsync method.
+        /// </summary>
+        /// <returns>A <see cref="UserCountsResponse"/> representing the response.</returns>
+        public virtual async Task<UserCountsResponse> GetCountsAsync()
+        {
+            return await _api.GetCountsAsync().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Wraps Slack API's GetDirectMessageHistoryAsync method.
+        /// </summary>
+        /// <param name="conversationInfo">.</param>
+        /// <param name="latest">The date of the latest.</param>
+        /// <param name="oldest">The date of the oldest.</param>
+        /// <param name="count">The requested count.</param>
+        /// <returns>A <see cref="MessageHistory"/> representing the retrieved message history.</returns>
+        public virtual async Task<MessageHistory> GetDirectMessageHistoryAsync(DirectMessageConversation conversationInfo, DateTime? latest = null, DateTime? oldest = null, int? count = null)
+        {
+            return await _api.GetDirectMessageHistoryAsync(conversationInfo, latest, oldest, count).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Wraps Slack API's GetDirectMessageListAsync method.
+        /// </summary>
+        /// <returns>A <see cref="DirectMessageConversationListResponse"/> representing the response to listing the messages.</returns>
+        public virtual async Task<DirectMessageConversationListResponse> GetDirectMessageListAsync()
+        {
+            return await _api.GetDirectMessageListAsync().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Wraps Slack API's GetFileInfoAsync method.
+        /// </summary>
+        /// <param name="fileId">The id of the file to retrieve the info from.</param>
+        /// <param name="page">The page number.</param>
+        /// <param name="count">The requested count.</param>
+        /// <returns>A <see cref="FileInfoResponse"/> representing the response.</returns>
+        public virtual async Task<FileInfoResponse> GetFileInfoAsync(string fileId, int? page = null, int? count = null)
+        {
+            return await _api.GetFileInfoAsync(fileId, page, count).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Wraps Slack API's GetFilesAsync method.
+        /// </summary>
+        /// <param name="userId">The user id.</param>
+        /// <param name="from">The initial date.</param>
+        /// <param name="to">The final date.</param>
+        /// <param name="count">The requested count.</param>
+        /// <param name="page">The page number.</param>
+        /// <param name="types">The type of files to retrieve.</param>
+        /// <returns>A <see cref="FileListResponse"/> representing the response.</returns>
+        public virtual async Task<FileListResponse> GetFilesAsync(string userId = null, DateTime? from = null, DateTime? to = null, int? count = null, int? page = null, FileTypes types = FileTypes.all)
```

Comment:
Why not create a data structure instead of creating multiple parameters?
Refer to [this post](https://stackoverflow.com/questions/333056/regarding-passing-many-parameters).

Reply:
I'm not sure. This would call for 13 new structures (some with repeated parameters) in the class to account for all the methods that receive multiple parameters as input